android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/BroadcastActivity.java

```diff
@@ -0,0 +1,205 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.physical_web.physicalweb;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.bluetooth.BluetoothAdapter;
+import android.bluetooth.BluetoothManager;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.util.Log;
+import android.util.Patterns;
+import android.widget.Toast;
+
+import org.physical_web.physicalweb.ble.UriBeacon;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+
+
+/**
+ * This is the main entry point for the broadcasting service.
+ * Handles the share intent.
+ */
+public class BroadcastActivity extends Activity {
+
+    private static final String TAG  = "BroadcastActivity";
+    private static final int ENABLE_BLUETOOTH_REQUEST_ID = 1;
+    public static final int MAX_URI_LENGTH = 18;
+    private String url;
+
+
+    @Override
+    protected void onActivityResult (int requestCode, int resultCode, Intent data) {
+        Log.d(TAG, "onActivityResult");
+        if (requestCode == ENABLE_BLUETOOTH_REQUEST_ID && resultCode == -1) {
+            checkBleAndStart();
+        } else {
+            Toast.makeText(this, getString(R.string.bt_on), Toast.LENGTH_LONG).show();
+        }
+        finish();
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        if (Build.VERSION.SDK_INT < 21){
+            Toast.makeText(this, getString(R.string.ble_os_error),
+                Toast.LENGTH_LONG).show();
+            return;
+        }
+        if (!checkIfBluetoothIsEnabled()) {
+            requestBluetooth();
+        } else {
+            checkBleAndStart();
+            finish();
+        }
+    }
+
+    private void parseAndHandleUrl(){
+        Intent intent = getIntent();
+        String action = intent.getAction();
+        String type = intent.getType();
+        String text = intent.getStringExtra(Intent.EXTRA_TEXT);
+        // Look to see if the text is a URL
+        url = findUrlInText(text);
+        // If the text is a URL
+        if (url != null) {
+            if (hasValidUrlLength(url) && isAsciiUrl(url)) {
+                // Set the url if we can
+                Log.d(TAG, "valid length");
+                startBroadcastService(url, url);
+            } else {
+                Log.d(TAG, "needs shortening");
```

Comment:
Yes, and those URLs still require "shortening" so that it will fit into the eddystone-url format